developer-docs-site/docs/guides/move-guides/move-scripts.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "Move Scripts"
+slug: "move-scripts"
+---
+
+# Move Scripts
+
+You can use Move scripts to conduct arbitrary tasks on chain. 
+
+## Example use case
+
+The following example calls functions on the [aptos_coin.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_coin.move) module to confirm the balance of the destination account is less than `desired_balance`, and if so, tops it up to `desired_balance`.
+
+```move
+script {
+    use std::signer;
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin;
+    use aptos_framework::coin;
+
+    fun main(src: &signer, dest: address, desired_balance: u64) {
+        let src_addr = signer::address_of(src);
+
+        let balance = coin::balance<aptos_coin::AptosCoin>(src_addr);
+        if (balance < desired_balance) {
+            aptos_account::transfer(src, dest, desired_balance - balance);
+        };
+
+        addr::my_module::do_nothing();
```

Comment:
nit: Instead of do_nothing(), how about naming it "do_something" so it's clear that this example shows how you can invoke custom logic?

Reply:
Actually, do we even need to show this custom call here instead of a simpler example where people call multiple existing functions in a Move script? Having module deployment in the instructions below makes it longer/more complex.

Reply:
I can make it `do_something` yeah. As for whether to have this, I found how to make it work with another module to be one of the most confusing parts of using Move scripts. I agree it makes it a bit more complex though, maybe it could be added as an extension afterwards.

Reply:
Yeah the simpler the better since we don't even have a simple script example. One possible way here is to initially make the script not call the custom module. Then add it at the end. This way users can go through and be able to run a script without the extra deploy step first. Then they can learn to do more advanced stuff by deploying the module

Reply:
For now I just removed it and at the end I say if they want to see how to use a user-defined Move module as well, they can go see my example code.

Reply:
Makes sense!

Reply:
I still see this here. Did you push the latest changes?

Reply:
Oop, just did it now.